src/main/java/com/exadel/discountwebapp/location/service/LocationService.java

```diff
@@ -0,0 +1,77 @@
+package com.exadel.discountwebapp.location.service;
+
+import com.exadel.discountwebapp.location.entity.Location;
+import com.exadel.discountwebapp.location.exception.LocationNotFoundException;
+import com.exadel.discountwebapp.location.mapper.LocationMapper;
+import com.exadel.discountwebapp.location.repository.LocationRepository;
+import com.exadel.discountwebapp.location.vo.LocationRequestVO;
+import com.exadel.discountwebapp.location.vo.LocationResponseVO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class LocationService {
+    private final LocationRepository locationRepository;
+    private final LocationMapper locationMapper;
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public List<LocationResponseVO> findAll() {
+        List<LocationResponseVO> response = new ArrayList<>();
+        locationRepository.findAll().forEach(en -> response.add(locationMapper.toVO(en)));
+        return response;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public LocationResponseVO findById(Long id) {
+        Optional<Location> location = Optional.ofNullable(locationRepository.findById(id)
+                .orElseThrow(() -> new LocationNotFoundException("Could not find location with id: " + id)));
```

Comment:
Should not be Optional<Location>

you can use Location location = locationRepository.findById(id)
                .orElseThrow(() -> new LocationNotFoundException("Could not find location with id: " + id))
                
orElseThrow will throw exception if Optional contains null, so location cannot be null

Reply:
thanks, Optional removed